src/Ilios/CoreBundle/Tests/Controller/AamcMethodControllerTest.php

```diff
@@ -0,0 +1,167 @@
+            ->getOne()['aamcMethod']
+        ;
+
+        $this->createJsonRequest(
+            'GET',
+            $this->getUrl(
+                'get_aamcmethods',
+                ['id' => $aamcMethod['id']]
+            )
+        );
+
+        $response = $this->client->getResponse();
+
+        $this->assertJsonResponse($response, Codes::HTTP_OK);
+        $this->assertEquals(
+            $aamcMethod,
+            json_decode($response->getContent(), true)['aamcMethod']
+        );
+    }
+
+    public function testGetAllAamcMethods()
+    {
+        $this->createJsonRequest('GET', $this->getUrl('cget_aamcmethods'));
+        $response = $this->client->getResponse();
+
+        $this->assertJsonResponse($response, Codes::HTTP_OK);
+        $this->assertEquals(
+            $this->container->get('ilioscore.dataloader.aamcmethod')->getAll(),
+            json_decode($response->getContent(), true)
+        );
+    }
+
+    public function testPostAamcMethod()
+    {
+        $this->createJsonRequest(
+            'POST',
+            $this->getUrl('post_aamcmethods'),
+            json_encode(
+                $this->container->get('ilioscore.dataloader.aamcmethod')
+                    ->create()['aamcMethod']
+            )
+        );
+
+        $response = $this->client->getResponse();
+        $this->assertEquals($response->getStatusCode(), Codes::HTTP_CREATED);
+        $this->assertArrayHasKey('location', $response->headers->all(), print_r($response->headers->all(), true));
+    }
+
+    public function testPostBadAamcMethod()
+    {
+        $invalidAamcMethod = array_shift(
+            $this->container->get('ilioscore.dataloader.aamcmethod')->invalid()
+        );
+
+        $this->createJsonRequest(
+            'POST',
+            $this->getUrl('post_aamcmethods'),
+            $invalidAamcMethod
+        );
+
+        $response = $this->client->getResponse();
+        $this->assertEquals($response->getStatusCode(), Codes::HTTP_BAD_REQUEST);
+    }
+
+    public function testPutAamcMethod()
+    {
+        $aamcMethod = $this->container
+            ->get('ilioscore.dataloader.aamcmethod')
+            ->createWithId()['aamcMethod']
+        ;
+
+        $this->createJsonRequest(
+            'PUT',
+            $this->getUrl(
+                'put_aamcmethods',
+                ['id' => $aamcMethod['id']]
+            ),
+            json_encode($aamcMethod)
+        );
+
+        $response = $this->client->getResponse();
+        $this->assertJsonResponse($response, Codes::HTTP_CREATED);
+        $this->assertEquals(
+            $this->container->get('ilioscore.dataloader.aamcmethod')
+                ->getLastCreated()['aamcMethod'],
+            json_decode($response->getContent(), true)['aamcMethod']
+        );
+    }
+
+    public function testDeleteAamcMethod()
+    {
+        $aamcMethod = $this->container
+            ->get('ilioscore.dataloader.aamcmethod')
+            ->createWithId()['aamcMethod']
```

Comment:
I don't like createWithId() this seems like a database concern not part of the dataLoaders.  Why not just use getOne() here?